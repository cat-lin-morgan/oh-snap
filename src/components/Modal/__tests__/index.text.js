import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';
import { photoPlural } from '../../../constants';

const mockCurrentPhoto = { ...photoPlural[0], index: 0};
const mockToggleModal = jest.fn();

afterEach(cleanup);

describe('modal component', () => {
    it('renders', () => {
         render(
            <Modal
            onClose={mockToggleModal}
            currentPhoto={mockCurrentPhoto}
            />
        )
    })

    it('matches snapshot', () => {
        const { asFragment } = render(
           <Modal
           onClose={mockToggleModal}
           currentPhoto={mockCurrentPhoto}
           />
       )

       expect(asFragment()).toMatchSnapshot();
   })
})

describe('Click Event', () => {
    it('call onClose handler', () => {
        const { getByTestId } = render(
            <Modal
            onClose={mockToggleModal}
            currentPhoto={mockCurrentPhoto}
            />
        );
        fireEvent.click(getByTestId('close-modal'));
        expect(mockToggleModal).toHaveBeenCalledTimes(2);
    });
})

//ffffffff