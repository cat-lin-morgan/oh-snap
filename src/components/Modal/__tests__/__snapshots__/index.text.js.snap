// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
    id="backdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Grocery aisle
      </h3>
      <img
        alt="Grocery aisle"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        data-testid="close-modal"
        id="close-button"
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
